mbuild/box.py

```diff
@@ -2,60 +2,99 @@
 
 import numpy as np
 
+from mbuild.coordinate_transform import AxisTransform, ChangeOfBasis
+from mbuild.exceptions import MBuildError
+
 __all__ = ['Box']
 
 class Box(object):
     """A box representing the bounds of the system.
 
-    Attributes
+    Parameters
     ----------
-    mins : np.ndarray, shape=(3,), dtype=float
-        Minimum x, y, z coordinates.
-    maxs : np.ndarray, shape=(3,), dtype=float
-        Maximum x, y, z coordinates.
-    lengths : np.ndarray, shape(3,), dtype=float
-        Box length in x, y and z directions.
-    angles : np.ndarray, shape(3,), dtype=float, default=[90,90,90]
-        Angles defining the tilt of the box
+    box_vectors : np.ndarray, shape=(3,3), dtype=float
+        Vectors that define a right-handed parallelpiped (Box).
+
 
+    Attributes
+    ----------
+    box_vectors : np.ndarray, shape=(3,3), dtype=float
+        Vectors that define the parallelpiped (Box).
+    Lx, Ly, Lz : float
+        Lengths of the Box in the x,y,z dimensions
+    xy,xz,yz : float
+        Tilt factors needed to displace an orthogonal box to its parallelpiped structure.
     """
```

Comment:
It seems like bookkeeping the box_vectors attribute will be difficult. Maybe internally you can only store the minimal degrees of freedom necessary for defining the box. I think that would be 6 variables: the three box lengths and the three angles, or you could do the Lx, Ly, Lz, and xy, xz, yz. Whatever ends up being easier to deal with. And then each of these will be its own attribute. On top of that you can have getters like `box_vectors` and `lengths` that would generate the respective arrays on the fly. And setters for the same things which will calculate the changes to the 6 attributes and set them on the fly. So then you would still be able to modify the lengths like:
`box.set_vectors(box.vectors * 5)`
or 
`box.set_vectors(box.vectors + [[2, 0, 0,], [0, 0, 0], [0, 0, 0]])`
What you wouldn't be able to do in a one-liner is this:
`box.vectors[2, 1] = 5`
you'd have to do:
```python3
new_vectors= box.vectors
new_vectors[2, 1] = 5
box.set_vectors(new_vectors)
```